src/native/pinvoke-override/pinvoke-override-api-impl.hh

```diff
@@ -19,7 +19,9 @@ namespace xamarin::android {
 		void *lib_handle = dso_handle == nullptr ? nullptr : *dso_handle;
 
 		if (lib_handle == nullptr) {
-			lib_handle = internal::MonodroidDl::monodroid_dlopen (library_name, MONO_DL_LOCAL, nullptr, nullptr);
+			// We're being called as part of the p/invoke mechanism, we don't need to look in the AOT cache
+			constexpr bool USE_AOT_CACHE = false;
```

Comment:
[Ditt](https://github.com/dotnet/android/pull/9117/files#r1731491238): this should probably be named `DO_NOT_USE_AOT_CACHE=false`.

Reply:
I changed the pattern by using an enum to parametrize the template instead.  Hopefully it's less confusing.